pkg/client/rp/relying_party.go

```diff
@@ -67,19 +67,26 @@ type RelyingParty interface {
 
 	// IDTokenVerifier returns the verifier used for oidc id_token verification
 	IDTokenVerifier() *IDTokenVerifier
-	// ErrorHandler returns the handler used for callback errors
 
+	// ErrorHandler returns the handler used for callback errors
 	ErrorHandler() func(http.ResponseWriter, *http.Request, string, string, string)
 
+	// UnauthorizedHandler returns the handler used for unauthorized errors
+	UnauthorizedHandler() func(http.ResponseWriter, *http.Request, string, string)
+
```

Comment:
Adding a method to the interface would be a breaking change. For the proposed implementation we don't need to expose the method, right?

Reply:
Correct, we do not need to expose the method. however CodeExchangeHandler is using the interface as input. Without declare the function on the interface, the method is not available.

https://github.com/zitadel/oidc/blob/2b35eeb83560d69c6b492eb5dcad02ad30c6da55/pkg/client/rp/relying_party.go#L448

As I mention, I would prefer to re-use the existing `rp.ErrorHandler`, but it does not fit in this case here.